File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/flutterobject/NativeObject.java
package com.valentingrigorean.arcgis_maps_flutter.flutterobject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.flutter.plugin.common.MethodChannel;

public interface NativeObject {
    String getObjectId();

    void dispose();

    void setMessageSink(@Nullable NativeMessageSink messageSink);

    void onMethodCall(@NonNull String method, @Nullable Object args, @NonNull MethodChannel.Result result);
}

File: android/src/main/kotlin/com/valentingrigorean/arcgis_maps_flutter/flutterobject/NativeObject.kt
package com.valentingrigorean.arcgis_maps_flutter.flutterobject

import io.flutter.plugin.common.MethodChannel

interface NativeObject {
    val objectId: String
    fun dispose()
    fun setMessageSink(messageSink: NativeMessageSink?)
    fun onMethodCall(method: String, args: Any?, result: MethodChannel.Result)
}